share/p2p/shrexeds/server.go

```diff
@@ -90,44 +93,44 @@ func (s *Server) handleStream(stream network.Stream) {
 	case errors.Is(err, eds.ErrNotFound):
 		status = p2p_pb.Status_NOT_FOUND
 	case err != nil:
-		log.Errorw("server: get car", "err", err)
+		logger.Errorw("server: get CAR", "err", err)
 		status = p2p_pb.Status_INTERNAL
 	}
 
 	// inform the client of our status
-	err = s.writeStatus(status, stream)
+	err = s.writeStatus(logger, status, stream)
 	if err != nil {
-		log.Warnw("server: writing status to stream", "err", err)
+		logger.Warnw("server: writing status to stream", "err", err)
 		stream.Reset() //nolint:errcheck
 		return
 	}
 	// if we cannot serve the EDS, we are already done
 	if status != p2p_pb.Status_OK {
 		err = stream.Close()
 		if err != nil {
-			log.Debugw("server: closing stream", "err", err)
+			logger.Debugw("server: closing stream", "err", err)
 		}
 		return
 	}
 
 	// start streaming the ODS to the client
-	err = s.writeODS(edsReader, stream)
+	err = s.writeODS(logger, edsReader, stream)
 	if err != nil {
-		log.Warnw("server: writing ods to stream", "hash", hash.String(), "err", err)
+		logger.Warnw("server: writing ods to stream", "err", err)
 		stream.Reset() //nolint:errcheck
 		return
 	}
 
 	err = stream.Close()
 	if err != nil {
-		log.Debugw("server: closing stream", "err", err)
+		logger.Debugw("server: closing stream", "err", err)
 	}
 }
 
-func (s *Server) readRequest(stream network.Stream) (*p2p_pb.EDSRequest, error) {
+func (s *Server) readRequest(logger *zap.SugaredLogger, stream network.Stream) (*p2p_pb.EDSRequest, error) {
```

Comment:
why?

Reply:
I want to use logger enriched with fields